overlord/devicestate/devicectx_test.go

```diff
@@ -0,0 +1,121 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package devicestate_test
+
+import (
+	"testing"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/boot/boottest"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/testutil"
+	. "gopkg.in/check.v1"
+)
+
+func TestBoot(t *testing.T) { TestingT(t) }
+
+type deviceCtxSuite struct {
+	testutil.BaseTest
+}
+
+var _ = Suite(&deviceCtxSuite{})
+
+func (s *deviceCtxSuite) SetUpTest(c *C) {
+}
+
+func testGroundDeviceContextCommonChecks(c *C, devCtx snapstate.DeviceContext) {
+	c.Check(devCtx.RunMode(), Equals, true)
+	c.Check(devCtx.GroundContext(), Equals, devCtx)
+	c.Check(func() { devCtx.Store() }, PanicMatches,
+		"retrieved ground context is not intended to drive store operations")
+	c.Check(devCtx.ForRemodeling(), Equals, false)
+	c.Check(devCtx.SystemMode(), Equals, "run")
+}
+
+func (s *deviceCtxSuite) TestGroundDeviceContext(c *C) {
+	var devCtx snapstate.DeviceContext
+
+	// Classic with classic initramfs
+	classModel := assertstest.FakeAssertion(map[string]interface{}{
```

Comment:
I wouldn't abbreviate classic to class, as class is a proper word as well with  a different meaning. I think is best to call this classicModel, same below